vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/TestRunResultsListener.kt

```diff
@@ -50,56 +54,85 @@ class TestRunResultsListener(private val testBatch: TestBatch,
         val results = mergeParameterisedResults(runResult.testResults)
         val tests = testBatch.tests.associateBy { it.identifier() }
 
-        val testResults = results.map {
-            it.toTestResult(device)
-        }
+        val passed = mutableListOf<TestResult>()
+        val failed = mutableListOf<TestResult>()
+        val infraFailures = mutableListOf<TestResult>()
+        val incomplete = mutableListOf<TestResult>()
 
-        val nonNullTestResults = testResults.filter {
-            it.test.method != "null"
-        }
+        results.forEach { entry ->
+            val result = entry.toTestResult(device)
+            if (result.test.method == "null") return@forEach
 
-        val finished = nonNullTestResults.filter {
-            results[it.test.identifier()]?.isSuccessful() ?: false
-        }
+            when (entry.value.status) {
+                DdmLibTestStatus.PASSED, DdmLibTestStatus.ASSUMPTION_FAILURE, DdmLibTestStatus.IGNORED -> passed.add(result)
```

Comment:
should `ASSUMPTION_FAILURE` be treated as passed?

Reply:
In my opinion, you are right. I've tried to find more info on the meaning of ASSUMPTION_FAILURE, but there is no much info. The only thing I can tell is that it has been this way for a long time and that it hasn't been changed in the main repo too. So let's keep it that way for now.